api/functions/generate_enums.py

```diff
@@ -40,9 +41,54 @@ def create_enums(Table, enum_name, column):
 		enum_dict.update({select[1]: select[1]})
 
 	if not len(enum_dict):
-		return graphene.Enum(enum_name, 'EMPTY')
+		ret_enum = enum.Enum(enum_name, 'EMPTY')
 	else:
-		return graphene.Enum(enum_name, enum_dict)
+		ret_enum = enum.Enum(enum_name, enum_dict)
+
+	return graphene.Enum.from_enum(ret_enum)
+
+
+def create_enum_with_descriptions(Table, enum_name, enum_column, description_column):
+	"""
+	Function to allow the creation of enums with descriptions for various uses. Function takes in SQLAlchemy Model,
+	and the Column that you would like your enums to be created. The value of the enum is equal to that of the name
+	"""
+	app = create_enum_app()
+	db = create_enum_db(app)
+
+	with app.app_context():
+		query = db.session.query(Table).options(load_only(enum_column))
+		rows = db.session.execute(query).fetchall()
+
+	enum_dict = dict()
+
+	for select in rows:
+		enum_dict.update({select[1]: select[1]})
+
+	if not len(enum_dict):
+		ret_enum = enum.Enum(enum_name, 'EMPTY')
+	else:
+		ret_enum = enum.Enum(enum_name, enum_dict)
+
+	description_dict = dict()
+
+	for e in ret_enum:
+		with app.app_context():
+			query = db.session.query(Table).filter(getattr(Table, enum_column) == e.name).options(load_only(description_column))
+			row = db.session.execute(query).first()
+
+			try:
+				description_dict.update({e.name: row[1]})
+			except:
+				description_dict.update({e.name: "Error loading description"})
```

Comment:
Could this go with the other error messages in `functions/error_messages.py`??

Reply:
This should not go into a function, if there is an error and the entire description column is empty and you have 1000+ rows that you are processing it will call that function the 1000+ on startup, this way it just sets a direct value